streaming/base/stream.py

```diff
@@ -417,26 +417,28 @@ def get_shards(self, world: World) -> List[Reader]:
         Returns:
             `List[Reader]: Shard readers.
         """
-        # Download the index.
+        # Download the index file if it does not exist locally.
         basename = get_index_basename()
         filename = os.path.join(self.local, self.split, basename)  # pyright: ignore
-        if world.is_local_leader:
-            if self.remote:
-                # Downloads the `index.json` as `index.json.tmp` fully and then rename it to
-                # `index.json` since only one process downloads the `index.json` file while
-                # other processes wait for it to get downloaded. Hence, It avoids loading the
-                # in-progress downloading `index.json`.
-                tmp_filename = self._download_file(basename, basename + '.tmp')
-                os.rename(tmp_filename, filename)
+        if not os.path.exists(filename):
+            if world.is_local_leader:
+                if self.remote:
+                    # Downloads the `index.json` as `index.json.tmp` fully and then rename it to
+                    # `index.json` since only one process downloads the `index.json` file while
+                    # other processes wait for it to get downloaded. Hence, It avoids loading the
+                    # in-progress downloading `index.json`.
+                    tmp_filename = self._download_file(basename, basename + '.tmp')
+                    os.rename(tmp_filename, filename)
+                else:
+                    if not os.path.exists(filename):
```

Comment:
You don't have to check if the path exists locally again right? just being in this block means the os path doesnt exist locally, so this condition might be redundant. other than this PR lgtm!

Reply:
This check is required when `remote=None, local=remote_dir` and `remote_dir` does not have an `index.json` file. I have added a test case for the same

Reply:
ah i see. cool!